### DISCUSSION TITLE
Mouse wheel click to close windows

### DISCUSSION THREAD

**Discussion by telk5093** [OpenTTD/OpenTTD#12888] [Features]
It will be great if we can close the windows by clicking mouse's wheel button, like web browsers instead of right click. (I have used to right clicking to show tooltips)

**Comment by LC-Zorg** [comment]
Considering the possibility of adding the infrastructure copying function, in the CityMania version, where it has been present for a long time, the mouse wheel button is used to rotate the copied element - maybe it's a matter of habit, but in my opinion this is the best location of this possibility and use of this button.

Besides, I can understand this idea due to habits and the desire to duplicate similar functionality from other places, but I completely do not understand the argument of getting used to displaying tooltips with the right button.

I have no idea where these tooltips are displayed under the right button. I turned off the close function specifically to check this and I do not see these hints anywhere. The only place is the vehicles in the depot, but there it also works with the window close function enabled. So where are these tooltips? Serious question. Can someone enlighten me? :)

To this day I don't understand why the right-click close feature, even though it makes night out of day, is still not enabled by default.

**Reply by telk5093** [reply]
I was talking about _clicking_ the mouse wheel button, not _rotating_ it.
Does CitiMania version using the mouse wheel button's clicking to rotate the copied element? (I've never used it so I have no idea)

And for more precise explanations, I'm using to show tooltips by this settings:
![image](https://github.com/user-attachments/assets/c24610ab-e2cc-4aed-9826-595e19a80daf)

Changing ``Close window on right-click`` setting to ``Close window`` and making more lists like ``right-click``, ``right-click, except sticky``, ``wheel-click``, ``wheel-click, except sticky`` and so on would be an option.

**Reply by LC-Zorg** [reply]
> Does CitiMania version using the mouse wheel button's clicking to rotate the copied element? (I've never used it so I have no idea)

Yes, that's also clicking the wheel.

Thanks for the explanation. That changes a lot. :)
I didn't see this setting, or rather I didn't know that by setting it to 0 I could get this effect. Although to tell you the truth I never cared about it. A long time ago I went the other way and increased the time to 1 second (those 0.25s and the constant appearance of this subtitles were annoying) and this is the best solution for me. I very, very rarely use these tips, and this 1 second means that they basically never appear unless I want them to. Their spontaneous appearance is even more convenient for me than having to click.

That's my point of view, but I think I understand your proposal now and it seems quite reasonable. There is still the mentioned conflict here and I don't know how popular the setting you use is, but looking at what is done more often, closing windows is definitely more common than copying and here this button would be much more useful. I don't know what IdpI thinks about it, but I think that if your proposal were implemented, for example, Spacebar could be assigned to rotate the copied infrastructure. I would do that and I think it wouldn't be a big loss for me. :) Certainly a smaller than the gain for you from the new opportunity. I can't imagine how you can play today while constantly closing windows with that little icon. ;)

By the way, most mice have more than 3 buttons... Maybe it would be worth adding the option of freely configuring all buttons? :) Nothing important, I'm just thinking out loud. ;)

If this closing function were added, or maybe even without it, I think it would be worth changing the "Show tooltips" setting. The current precise setting of every 1 millisecond is unnecessary and even strange in my opinion. Meanwhile, in the case of a gradual setting, not only would the choice be easier, but the "right click" option would also be much more visible. Because not every player reads explanations. ;) It could be... 

Show tooltips: 
- after 0.25s 
- after 0.5s 
- after 0.75s 
- after 1s 
- after 1.5s 
- after 2s 
- under right mouse button

**Reply by ldpl** [reply]
I don't use right click to close, only close all hotkey, so have no particular opinion on that. 
Just want to add that in CityMania client middle mouse button (aka wheel) is used as a general rotate/change layout function in multiple places: rotating blueprints, depots and road stops, flipping stations and changing industry layout when funding. And I consider that to be one of the most useful features of cmclient. Also, it's already configurable in hotkeys.cfg

**Comment by nielsmh** [comment]
I think being able to close windows with a click other than on the designated Close button is a misfeature is general. I don't want to accidentally close windows. Having it as a selectable setting is also just bound to cause confusion.

One thing I think might be acceptable is having a keyboard hotkey to close the window last clicked inside. There is the Delete key to close all non-stickied windows, Shift-Delete to close all windows including stickied, Ctrl-Delete to close all message windows. As far as I remember, Rollercoaster Tycoon and/or Locomotion came up with an additional key: Backspace to close one window.
Why not add that? Backspace to close the last clicked-inside window.

**Reply by ldpl** [reply]
IMO the main issue here is the window spam, not the lack of ways to close them. Though fixing that, of course, is much more complex.

**Reply by nielsmh** [reply]
Fixing window spam properly then, would need to involve either opening fewer windows, somehow, or introducing some kind of temporary windows? Like only one vehicle or station window open at a time, unless you pin them, or similar logic.
